### ISSUE TITLE
Upgrade to MongoDB 3

### ISSUE THREAD

**Issue by steveswinsburg** [orionhealth/XBDD#37]
Upgrade the driver code:
http://mongodb.github.io/mongo-java-driver/3.0/whats-new/upgrading/

Optional: Write doco on upgrading the database:
https://askubuntu.com/questions/643252/how-to-migrate-mongodb-2-6-to-3-0-with-wiredtiger

**Comment by rushton3179** [comment]
I have done some investigating and this while we can just upgrade to mongodb 3 and consequently version 3 of the mongodb java driver there is some work that should be done. Specifically the mongo.getDB() method is deprecated. 

Upgrading this to getDatabase() then begins a cascade of deprecated code. The biggest from this I have found is the deprecation of the JSON class. Now instead the BasicDBObject parses JSON string directly into BasicDBObject's using its own parse method. 

Before JSON class would parse JSON lists into the BasicDBList type of which there is now no method to do so, making this the biggest obstacle. I have experimented with doing this via googles gson library but the conversion is never appropriate. I believe it would be better to change how we handle this case, opposed to forcing the data into the old form.
